tools/install_synapse.sh

```diff
@@ -1,13 +1,20 @@
 #!/usr/bin/env bash
 
-set -ex
+set -exo pipefail
+
+PYTHON2_VERSION=$(python2 -c 'import sys; print ".".join(str(v) for v in sys.version_info[:2])' || true)
+
+if [[ ${PYTHON2_VERSION} != "2.7" ]]; then
+    echo This script requires Python 2.7
+    exit 1
+fi
 
 SYNAPSE_URL="${SYNAPSE_URL:-https://github.com/matrix-org/synapse/tarball/master#egg=matrix-synapse}"
 SYNAPSE_SERVER_NAME="${SYNAPSE_SERVER_NAME:-matrix.local.raiden}"
-BASEDIR=$(readlink -m "$(dirname $0)/..")
+BASEDIR=$(python3 -c 'import sys; from pathlib import Path; print(Path(sys.argv[1]).parent.parent.absolute())' "$0")
 
-if [ ! -d "${DESTDIR}" ]; then
-    if [ -n "${TRAVIS}" ]; then
+if [[ ! -d ${DESTDIR} ]]; then
+    if [[ -n ${TRAVIS} ]]; then
```

Comment:
does the builtin `[[` handle variable expansion differently?  IOW, removing the `"` wont possibly result in `n` arguments instead of 1?

Reply:
Yes, that's the big advantage of using `[[` over `[`. You don't need to worry about `$IFS`, prefixing zero length strings and all that nonsense. ([Bash manual section on `[[`](http://www.gnu.org/software/bash/manual/bashref.html#index-_005b_005b))

Reply:
Yeah that's why I always try to use `bash` over `sh, to take advantage of `[[`